ai21/clients/common/beta/assistant/plans.py

```diff
@@ -15,20 +19,33 @@ def create(
         self,
         *,
         assistant_id: str,
-        code: str,
+        code: str | callable,
+        schemas: list[dict] | list[Type[BaseModel]] | NotGiven = NOT_GIVEN,
         **kwargs,
     ) -> PlanResponse:
         pass
 
     def _create_body(
         self,
         *,
-        code: str,
+        code: str | callable,
+        schemas: list[dict] | list[Type[BaseModel]] | NotGiven = NOT_GIVEN,
         **kwargs,
     ) -> Dict[str, Any]:
+        if callable(code):
+            code = inspect.getsource(code).strip()
+
+        schema_dicts = []
+        for schema in schemas:
+            if inspect.isclass(schema) and issubclass(schema, BaseModel):
+                schema_dicts.append(schema.model_json_schema())
+            else:
+                schema_dicts.append(schema)
```

Comment:
Can you please move this to a protected function? so the usage would be something like - 
```python
schema_dicts = [self._parse_schema(schema) for schema in schema]
```